README.md

```diff
@@ -163,7 +163,7 @@ fn write_io() -> Result<(), scroll::Error> {
     let mut bytes = [0x0u8; 10];
     let mut cursor = Cursor::new(&mut bytes[..]);
     cursor.write_all(b"hello")?;
-    cursor.iowrite_with(0xdeadbeef as u32, BE)?;
+    cursor.iowrite_with(&(0xdeadbeef as u32), BE)?;
```

Comment:
This and the 3 others are a huge PR, will take me a bit to review, appreciate your patience, @willglynn.
I would like to note that _having_ to take the reference now of variables, literals, makes me pretty sad...
I feel like it jars slightly with the original and basic purpose of scroll, which was to "serialize" numbers and PODs.
I do wish there had been a little more discussion before this enormous (and amazing PR!), specifically, whether we can have our cake and eat it too w.r.t. TryInto for the primitives.
Specifically, I'm wondering if there's a compromise, so that we can still pass either references _or_ owned values to the pwrite/iowrite.
Maybe it's just me but i do find it a usability paper cut and find it less attractive that I have to take references to pwrite now, but perhaps I'm used to the old way.
So, is there some generic magic we can use to do this? Guessing wildy and randomly:
1. Have someone take AsRef
2. Have scroll_derive derive the reference versions for any user type (i'm ok with taking references of user structs)
3. ???

On the other hand, in my case where the struct was too large to be put on the stack, and I had to box it, even if we did say 2., this would still probably eventually blow the stack? Or perhaps not. Have to think about it, and its late and a Thursday :)
Similarly, and against my original sadness, if I'm able to pwrite out a huge boxed struct that auto-derives pwrite, that would actually make me happy and would counteract any bitterness of no longer being able to pwrite raw deadbeefs :D
Dunno, just some thoughts, feel free to respond whenever you get the time (and that goes for anyone else who's watch).
Regardless, thanks for your amazing work here @willglynn and the robustness and tenacity of the changes, great work!

Reply:
Cheers! And yeah, this is why I went and did these PRs even with limited discussion: I didn't know how this concept would look until it was done 😄

I'll ponder the generics question. There might be a way we can make `pwrite()` be `&`-agnostic while keeping `TryIntoCtx` take `&self` – that'd let you `pwrite()` either by value or by reference without consuming `self` during the write.